specification/securityinsights/resource-manager/Microsoft.SecurityInsights/preview/2021-09-01-preview/AlertRules.json

```diff
@@ -1338,6 +1435,9 @@
             {
               "$ref": "#/definitions/AlertRuleTemplatePropertiesBase"
             },
+            {
+              "$ref": "#/definitions/QueryBasedAlertRuleTemplateProperties"
+            },
             {
               "$ref": "#/definitions/ScheduledAlertRuleCommonProperties"
             }
```

Comment:
These kind of unnamed schema and multiple "allOf" section nested multiple layers usually not working well with SDK. If you don't need the SDK then they are fine.

Reply:
@weidongxu-microsoft ,
As far as we understood, these multiple "allOf"s in the properties object do not create inheritance but union of those properties. The purpose is to avoid code duplication(instead of duplicate those common fields in several places).
We've generated C# classes and it looks fine.

Reply:
In this particular case, `x-ms-client-flatten` flatten all these hierarchies into the flat `ScheduledAlertRuleTemplate` properties, so you are probably safe. But in other case you are likely getting meaningless class name or breaking changes on class name in future.

This is a suggestion. You are OK to ignore, as your are in preview version anyway.